### PR TITLE
Document where Polynomial parameters are optional

### LINKED ISSUE #4018: Optional parameters in Polynomial models should be indicated as such
In the polynomial models, all parameters other than 'degree' are optional and should be indicated as such:

http://docs.astropy.org/en/stable/_modules/astropy/modeling/polynomial.html#Polynomial2D

for example ``x_domain`` should be:

```
x_domain : list or None, optional
```

### PR DESCRIPTION
**Pull request by swt30**
This quick PR adds "optional" to the docs where the Polynomial classes have optional parameters. I see @embray might be updating these classes anyway, so feel free to close this if you've got it under control. 

Closes #4018

### CODE REVIEW

**Comment by swt30** [comment]
Sorry, I should have a put a [skip ci] in somewhere there since it's a documentation-only change.

File: astropy/modeling/polynomial.py [hunk 1]
```diff
@@ -165,15 +165,15 @@ class OrthoPolynomialBase(PolynomialBase):
...
+    x_window : list or None, optional
         range of the x independent variable
-    y_domain : list or None
+    y_domain : list or None, optional
         domain of the y independent variable
-    y_window : list or None
+    y_window : list or None, optional
         range of the y independent variable
-    param_dim : int
+    param_dim : int, optional
```
Comment (embray) [inline astropy/modeling/polynomial.py]: This argument is actually deprecated.  I'd just remove the reference to it outright.  The correct argument to use is `n_models` which is mentioned in the `Model` base class, I think.  The Model doctrings still need broader reworking in order to better document the optional keyword arguments that are common to most model classes.

**Comment by embray** [comment]
Turns out I was wrong about `n_models` being documented in the base `Model.__init__`.  Could you update this PR by removing the references to param_dim?  I'll add the documentation for the other arguments that are missing from the `Model` docstring.

**Comment by swt30** [comment]
All done - sorry for the delay!

**Comment by embray** [comment]
Thanks! I'll merge this in a sec--I just want to leave it open a bit longer as a reminder to myself to improve the `Model.__init__` docs.

**Pull request merged**